### ISSUE TITLE
Content-Type audio/mp4 crashes StreamingKit

### ISSUE THREAD

**Issue by patrickjquinn** [tumtumtum/StreamingKit#325]
Upon attempting to play an `audio/mp3` stream url streamingKit throws an exception (generic, bad exec, debugging simply gives me a memory address)  on line `2625` of `STKAudioPlayer.m ` which is `status = AudioConverterFillComplexBuffer(audioConverterRef, AudioConverterCallback, (void*)&convertInfo, &framesToDecode, &localPcmBufferList, NULL);`

Here is a sample `audio/mp4` file https://r10---sn-q0c7dn7z.googlevideo.com/videoplayback?initcwndbps=1462500&ip=2a02%3A8084%3A601d%3A600%3A885b%3Aaa1a%3Acac3%3Adbd0&requiressl=yes&pl=30&beids=%5B9452307%5D&source=youtube&keepalive=yes&mime=audio%2Fmp4&expire=1487877216&itag=140&sparams=clen%2Cdur%2Cei%2Cgcr%2Cgir%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Ckeepalive%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cpl%2Crequiressl%2Csource%2Cupn%2Cexpire&dur=138.530&mm=31&mn=sn-q0c7dn7z&gir=yes&clen=2200833&lmt=1480834295190049&ms=au&mt=1487855425&mv=m&ipbits=0&key=yt6&gcr=ie&ei=AOCuWKHoIMHKcoqGkIgJ&id=o-AJVXRqJwoCwcHNw7ViqBkGEJU08iuOxzZ1hjFrdC2f7C&upn=i9yFWaR38kA&signature=5CB79DB5EA4720D6BA1617E0CE9E531B4DB0A844.BFAC0F042634DCC1791CB4B7F5CD52FE521F1D88&ratebypass=yes (I'm going to remove this link in a day or so after someone tests it on the grounds of potential copyright concerns)

**Comment by patrickjquinn** [comment]
Okay @tumtumtum I know you're not around much anymore but if you have even 2 minutes I'd really appreciate you taking a look at this with me! 

The error is occurring on line `2625` of `STKAudioPlayer.m` i.e:   `status = AudioConverterFillComplexBuffer(audioConverterRef, AudioConverterCallback, (void*)&convertInfo, &framesToDecode, &localPcmBufferList, NULL);`

Any help on this would be greatly appreciated.

**Comment by tumtumtum** [comment]
I will try to look at this this week. My Mac is in for repairs :-(

Is it possible the format of the file is incorrect or not encoded for streaming?

**Comment by patrickjquinn** [comment]
Hey @tumtumtum thank for getting back to me! It is possible that this is the case but it streams fine in VLC and via the browser (i.e its headers accept range, its content type isn't application:stream-octect etc) 

But more interestingly it doesn't play once its downloaded locally to the device and fed in as a file path url (This applies to all streams of content type audio/mp4, not just the one above). 

I've been seeing complaints generally on iOS that H264 encoded files no longer play properly as of iOS 9+, this could have something to do with it?

**Comment by patrickjquinn** [comment]
@tumtumtum Here is a sample bit of audio encoded as audio/mp4 that causes the crash (Its an open source song so not copyright conflicts here)

[2F904F2F-39E5-4492-8C02-6D5D700F04BC-23385-000A0B42ADCAC2E2.mp4.zip](https://github.com/tumtumtum/StreamingKit/files/844418/2F904F2F-39E5-4492-8C02-6D5D700F04BC-23385-000A0B42ADCAC2E2.mp4.zip)

**Comment by patrickjquinn** [comment]
Hey @tumtumtum did you get a chance to look at this yet? I've a brand new shiny app that I'm waiting to push live to the App Store with this as the only blocker (I've at least 4 apps live which use your framework so I'm a diehard supporter of your work) :)

**Comment by patrickjquinn** [comment]
Alright, i've had to move my project, lock stock, over to https://github.com/StreetVoice/HysteriaPlayer which supports this content type properly(ish)

If ever you fix this i'll migrate back to StreamingKit.

Let me know if I can help when you get around to it.

**Comment by patrickjquinn** [comment]
Hey @tumtumtum did you ever get your mac back and have a chance to look into this issue? I've zero problems working through this bug with you helping you debug and resolve it!

**Comment by kiwi07** [comment]
Hello Patrick,

Have you solved this issue?
I have the same problem with that kind of url.

**Comment by kiwi07** [comment]
Btw, hysteriaplayer cannot play the url too.

**Comment by patrickjquinn** [comment]
Hysteria player can in fact handle this data source, I switched everything over to it and it worked fine, I decided to implement my own player using AVPlayer instead of AVQueuePlayer. Core concept is the same. 

Am I right in saying that your content is coming from YouTube using the 140 tag?

**Comment by kiwi07** [comment]
Hi Patrick,

Yes, you're right, I got the streaming url from youtube like yours. However, Hysteria cannot stream that link. I will send you the log tomorrow, Please tell me in detail how to make it works. Unfortunately, Avplayer takes very long time to start to play the content.

**Comment by patrickjquinn** [comment]
Yes thats going to be your biggest problem. Like 10-30 second delays in some cases.... If you show me how you're doing it i'll do my best to resolve your problem. 

Ideally if someone could find a solution for the issue with StreamingKit i'd switch back to that but until then we will have to compromise.

**Comment by kiwi07** [comment]
:D i have the same issue. It delays 10-20 seconds before starting plays a content that has long duration. According to the Hysteria, I just paste the url in the sample code to test this pod.

**Comment by patrickjquinn** [comment]
Probably best that we continue this over on the Hysteria GitHub repo instead of here, dont want to spam @tumtumtum

**Comment by kiwi07** [comment]
Please send me your email. I would like to ask you for detail about hysteria.

**Comment by kiwi07** [comment]
Hello Patrick,

I've tested on HysteriaPlayer with the url like this: https://redirector.googlevideo.com/videoplayback?id=f61e2c1fe780f8b5&itag=140&source=youtube&requiressl=yes&initcwndbps=2528750&mm=31&ei=xwYdWYPfHsra-gOewJCoAQ&mv=m&ms=au&pl=20&mn=sn-n4v7sne7&ratebypass=yes&mime=audio/mp4&gir=yes&clen=98697746&lmt=1492920570151300&dur=6214.240&upn=7QyTJ-2gRgU&signature=761F17A80B11380EA7455D7DD6F38A6142915370.94FACF4B67323C8DAE455374E293099494D1329B&mt=1495074384&key=dg_yt0&ip=45.63.90.180&ipbits=0&expire=1495096103&sparams=ip,ipbits,expire,id,itag,source,requiressl,initcwndbps,mm,ei,mv,ms,pl,mn,ratebypass,mime,gir,clen,lmt,dur

it delays about 10s-15s before playing. How can I start streaming immediately?

Thank you for your time.

**Comment by marf** [comment]
Hello @patrickjquinn  and @kiwi07 I have the exact same problem, have you found a solution for it?

Thank you,
Marco
